upgrade-tests/src/test/java/io/zeebe/test/UpgradeTest.java

```diff
@@ -7,453 +7,134 @@
-          }
-        });
+  @AfterAll
+  static void tearDown() {
+    Optional.ofNullable(network).ifPresent(Network::close);
   }
 
-  @Ignore
-  @Test
-  public void oldGatewayWithNewBroker() {
+  @Timeout(value = 5, unit = TimeUnit.MINUTES)
+  @ParameterizedTest(name = "{0}")
+  @ArgumentsSource(UpgradeTestCaseProvider.class)
+  void oldGatewayWithNewBroker(final String name, final UpgradeTestCase testCase) {
     // given
-    state
-        .broker(CURRENT_VERSION, tmpFolder.getRoot().getPath())
-        .withStandaloneGateway(LAST_VERSION)
-        .start(true);
+    state.broker(CURRENT_VERSION).withStandaloneGateway(LAST_VERSION).start(true);
     final long wfInstanceKey = testCase.setUp(state.client());
 
     // when
     final long key = testCase.runBefore(state);
 
     // then
     testCase.runAfter(state, wfInstanceKey, key);
-    TestUtil.waitUntil(() -> state.hasElementInState(PROCESS_ID, "ELEMENT_COMPLETED"));
+    awaitProcessCompletion();
   }
 
-  @Ignore("https://github.com/zeebe-io/zeebe/issues/5385")
-  @Test
-  public void upgradeWithSnapshot() {
-    upgradeZeebe(true);
+  @Timeout(value = 5, unit = TimeUnit.MINUTES)
+  @ParameterizedTest(name = "{0}")
+  @ArgumentsSource(UpgradeTestCaseProvider.class)
+  void upgradeWithSnapshot(final String name, final UpgradeTestCase testCase) {
+    upgradeZeebe(testCase, true);
   }
 
-  @Test
-  public void upgradeWithoutSnapshot() {
-    upgradeZeebe(false);
+  @Timeout(value = 5, unit = TimeUnit.MINUTES)
+  @ParameterizedTest(name = "{0}")
+  @ArgumentsSource(UpgradeTestCaseProvider.class)
+  void upgradeWithoutSnapshot(final String name, final UpgradeTestCase testCase) {
+    upgradeZeebe(testCase, false);
   }
 
-  private void upgradeZeebe(final boolean withSnapshot) {
+  private void upgradeZeebe(final UpgradeTestCase testCase, final boolean withSnapshot) {
     // given
-    state.broker(LAST_VERSION, tmpFolder.getRoot().getPath()).start(true);
+    state.broker(LAST_VERSION).start(true);
     final long wfInstanceKey = testCase.setUp(state.client());
     final long key = testCase.runBefore(state);
 
     // when
-    final File snapshot = new File(tmpFolder.getRoot(), "raft-partition/partitions/1/snapshots/");
-
     if (withSnapshot) {
-
-      state.close();
-
-      state.broker(LAST_VERSION, tmpFolder.getRoot().getPath()).start(false);
-
-      // since 0.24, no snapshot is created when the broker is closed
-      Awaitility.await()
-          .atMost(Duration.ofMinutes(2))
-          .untilAsserted(
-              () ->
-                  assertThat(snapshot)
-                      .describedAs("Expected that a snapshot is created")
-                      .exists()
-                      .isNotEmptyDirectory());
-
+      // it's necessary to restart without the debug exporter to allow snapshotting
       state.close();
+      state.broker(LAST_VERSION).start(false);
+      EitherAssert.assertThat(state.getPartitionsActuatorClient().takeSnapshot()).isRight();
```

Comment:
I'm not adding a change request because it's just an opinion but I don't think this is very readable. The assert `isRight` doesn't look good to me but I don't have a good alternative so we can leave it. Maybe `hasRight` would also be misleading because it sounds like it can have both right and left. I think extracting the value would make it more obvious what the assert is actually checking though:
```suggestion
      Either<Throwable, Map<String, PartitionStatus>> response = state.getPartitionsActuatorClient().takeSnapshot();
      EitherAssert.assertThat(response).isRight();
```

Reply:
Hm, I think that's just a fundamental problem with a generic structure like `Either` =/ I admit in this case I'd rather have something like Rust's [Result](https://doc.rust-lang.org/std/result/) type, where you would then write something like `isOk()` (since it's either OK or failed, which is really what I use it as for the client).

Reply:
I added a description to the assert via `as()` for better error message :+1: